lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -54,62 +53,47 @@ def __init__(self, subsection, course):
 
         self.graded_total = None  # aggregated grade for all graded problems
         self.all_total = None  # aggregated grade for all problems, regardless of whether they are graded
-        self.locations_to_weighted_scores = OrderedDict()  # dict of problem locations to (Score, weight) tuples
-
-        self._scores = None
+        self.locations_to_scores = OrderedDict()  # dict of problem locations to ProblemScore
 
     @property
     def scores(self):
         """
         List of all problem scores in the subsection.
         """
-        if self._scores is None:
-            self._scores = [score for score, _ in self.locations_to_weighted_scores.itervalues()]
-        return self._scores
+        return self.locations_to_scores.values()
 
-    def init_from_structure(self, student, course_structure, scores_client, submissions_scores):
+    def init_from_structure(self, student, course_structure, submissions_scores, csm_scores):
```

Comment:
Good renaming.  Makes the relationship between scores_client and submission_scores clearer.